client/jetpack-onboarding/main.jsx

```diff
@@ -84,6 +88,20 @@ class JetpackOnboardingMain extends React.PureComponent {
 		}
 	}
 
+	setSelectedSite() {
+		const { hasFinishedRequestingSite } = this.state;
+		const { isConnected, selectedSiteId, siteId } = this.props;
+
+		if ( ! hasFinishedRequestingSite ) {
+			return;
+		}
+
+		if ( isConnected && selectedSiteId !== siteId ) {
```

Comment:
Do we need the `selectedSiteId !== siteId` check? Having both a `siteId` and a `selectedSiteId` prop coming from the selector can be quite confusing, especially for developers extending JPO later. There's a rather high risk they might not just use the wrong one.

While it's motivated differently here, it reminds me of the many legacy cases where a component gets e.g. `site`, `selectedSiteId`, and/or `siteId` props, and a dev making a drive-by change often just picks whatever they see first...

Reply:
We need the check because we don't want to be setting the selected site again on every prop change.

And while I understand the rationale for your doubt, I think the comment on the following line explains why we're doing this.

Reply:
> And while I understand the rationale for your doubt, I think the comment on the following line explains why we're doing this.

It's enough for us, since we know the context of JPO, unconnected sites, and all that. It's not enough to deter a drive-by dev from using `selectedSiteId` instead of `siteId` tho, especially when they're conditioned to look out for that name, as they maybe would in other components.

So we either need moar commentz (and possibly renaming of `selectedSiteId` to something really ugly 😁), or we could move the `selectedSiteId !== siteId` check to `connect` and pass a boolean prop (e.g. `isSiteSelected`) instead of `selectedSiteId` to the component -- that way, things would be a bit more contained. I think that latter idea would have me more at ease; not totally sure it'd work with the lifecycle methods we need, but I'm fairly optimistic it would (since we're nowhere using `nextProps` or `prevProps`).

Reply:
> renaming of selectedSiteId to something really ugly

Why? This is actually the real selected site, so I don't see a good reason to rename it.

I'm happy with the current logic because it's simple enough, but would be willing to add some more comments to make it more comprehensive. 

Could you give me some feedback on which bits you feel are fuzzy so I can document them further?

Reply:
Scenario:

* It's 2019
* Someone wants to add a feature to JPO
* They see `main` has `selectedSiteId` and recognize that from other components as "the currently selected/relevant `siteId`". They don't look any further, thus ignoring `siteId`.
* They add some new feature/check to `main`, using `selectedSiteId`.
* They might even test it with their connected wp sandbox, where it works.
* However, things break for unconnected sites.

I'd just really like to prevent people from mistaking `selectedSiteId` for "the currently relevant `siteId`", which is what a lot of people will expect from that name, but which in case of JPO it plainly isn't -- not if the site's not connected. As I said, I've seen this sort of confusion/mess too often with components that have e.g. both a `site` object and a `selectedSiteId` prop.

Reply:
Thanks for the example, I see why this could be a problem.

Got any good recommendations for renaming `selectedSiteId` in that case?

Reply:
I think my preferred way would be to do `isSiteSelected = selectedSiteId !== siteId` in  `connect()` and only pass that as a prop.

If you'd rather keep it in the actual component... hmm. Naming is hard. `selectedConnectedSiteId`? `selectedSiteIdWhenConnected`? Tricky. We want deterringly-ugly-but-still-meaningful 😆

Reply:
👍 Followed your first suggestion in https://github.com/Automattic/wp-calypso/pull/23293/commits/772d246bc9328e764dce7a940dcf422aa5bd12f9.